src/cargo/ops/cargo_new.rs

```diff
@@ -141,28 +297,90 @@ fn mk(config: &Config, path: &Path, name: &str,
         (Some(name), None, _, None) |
         (None, None, name, None) => name,
     };
+    
+    let mut cargotoml_path_specifier = String::new();
+    
+    let mut there_was_already_a_lib = false;
+    let mut previous_lib_relpath = String::new();
+    
+    // Calculare what [lib] and [[bin]]s do we need to append to Cargo.toml
+    
+    let explicit_better_than_implicit = opts.source_files.len() > 1;
```

Comment:
Hm, so I guess the logic I'd expect is something along the lines of:

* If the directory has a layout that matches standard Cargo idioms, no warnings or errors should be issued. This includes the ability to have multiple binaries with one library (e.g. `src/bin` + `src/lib.rs`)
* If a nonstandard layout is encountered, at most one binary and one library can be found. This is because Cargo only allows one library and otherwise we don't know where to find multiple binaries.

Does that make sense?